### PR TITLE
Initial work on OSD architecture

### PR DESCRIPTION
**Pull request by heitorPB**
This is an initial work to document the architecture used in the charms.

Wainting for #8 to be merged

### CODE REVIEW

File: operations/infiniband.rst [hunk 1]
```diff
@@ -0,0 +1,34 @@
+.. _infiniband:
+
+==========
+Infiniband
+==========
+
+The ``slurmd`` charm have a few actions to manage the Infiniband drivers. A
```
Comment (jamesbeedy) [inline operations/infiniband.rst:7]:
```suggestion
The ``slurmd`` charm has a few actions to manage the Infiniband drivers. A
```

File: architecture.rst [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+On the compute nodes, we also install
+`Node Health Check (NHC) <https://github.com/mej/nhc>`_ with a only a few basic
+checks for the node health status. It is possible, and recommended, that the
+cluster administrator extends these checks. Check :ref:`nhc` section for
+details on how to configure it.
+
+The easiest way to install Infiniband drivers on the compute nodes is to use
+the charm supplied actions related to infiniband management. Check
+:ref:`infiniband` section for details on how to install it.
```
Comment (jamesbeedy) [inline architecture.rst:29]:
```suggestion
:ref:`infiniband` section for details on Infiniband driver lifecycle operations.
```

File: operations/nhc.rst [hunk 3]
```diff
@@ -0,0 +1,53 @@
+.. _nhc:
+
+===
+NHC
+===
+
+OSD installs `LBNL Node Health Check (NHC) <https://github.com/mej/nhc>`_ by
+default in the ``slurmd`` charm. The base configuration file contains only
+basic checks, and you can easily extend it to match your own cluster.
```
Comment (jamesbeedy) [inline operations/nhc.rst:9]:
```suggestion
basic checks to ensure the slurm and munged proccese are active. You can easily extend the nhc configuration to match your setup.
```

File: operations/nhc.rst [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+   * || check_hw_ib 100
+
+And then configure your ``slurmd`` application to use it:
+
+.. code-block:: bash
+
+        $ juju config slurmd nhc-conf="$(cat custom-conf.nhc)"
+
+Note that this will append your custom configurations to the ``nhc.conf`` file,
+without overwriting the file.
```
Comment (jamesbeedy) [inline operations/nhc.rst:27]:
```suggestion
without overwriting the pre-existing checks defined by the charm.
```

File: index.rst [hunk 5]
```diff
@@ -5,17 +5,32 @@
...
+are the components that encapsulate the operational know-how and automation
+needed to facilitate the lifecycle of a SLURM cluster.
 
-   Slurm Installation Overview <slurm-installation/overview>
-   Operations Overview <operations/overview>
+Slurm Bundles
+-------------
+The `slurm-bundles <https://github.com/omnivector-solutions/slurm-bundles/>`_
+define the base slurm deployment configurations for different clouds and
+operating systems.
```
Comment (heitorPB) [inline index.rst:23]: What do you think about moving this to the Architecture page?

File: architecture.rst [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+are the components that encapsulate the operational know-how and automation
+needed to facilitate the lifecycle of a SLURM cluster.
+
+Slurm Bundles
+-------------
+The `slurm-bundles <https://github.com/omnivector-solutions/slurm-bundles/>`_
+define the base slurm deployment configurations for different clouds and
+operating systems.
+
+
```
Comment (heitorPB) [inline architecture.rst:24]:
```suggestion
OSD Components
--------------
```

Comment (heitorPB) [reply architecture.rst:24]: I'm not sure `components` is  good word here...

Comment (heitorPB) [reply architecture.rst:24]: The way it is now suggests the components and NHC are part of the bundles, that's why i suggest to add a new subsection.

**Comment by jamesbeedy** [comment]
👍 merge it!

**Pull request merged**